interfaces.go

```diff
@@ -79,6 +79,9 @@ type Method interface {
 
 	// Stores the next location to evaluate in xNext
 	Iterate(loc *Location, xNext []float64) (EvaluationType, IterationType, error)
+
+	// Requires returns the minimum requirements of Method on Function.
```

Comment:
I think this could use a more elaborate comment. For example, there isn't a direct correspondance between the requirements and the methods (you can have FuncGrad without Grad). It's tricky enough that a couple sentences are okay. "Requires specifies information about the objective function needed by the optimizer beyond just the function value. For example, some optimizers must have the gradient ... "